Lib/typing.py

```diff
@@ -486,9 +486,7 @@ def _evaluate(self, globalns, localns):
                 globalns = localns
             elif localns is None:
                 localns = globalns
-            self.__forward_value__ = _type_check(
-                eval(self.__forward_code__, globalns, localns),
-                "Forward references must evaluate to types.")
+            self.__forward_value__ = eval(self.__forward_code__, globalns, localns)
```

Comment:
I am not sure just removing the check is the right thing to do. Instead I would rather fix `_type_check`.

We can of course abandon any runtime validation, but this should be a conscious decision and goes beyond the scope of this patch.

Reply:
Well, `get_type_hints()` is not running _type_check on non-forward references so I feel it's a bit weird that we do it for forward references only. Why do we need the additional check for that case? It did successfully evaluate after all, just as if you didn't use a ForwardRef.

Reply:
This is indeed inconsistent, but the proposed solution creates a bigger inconsistency IMO. This code is also called from other places like `List['C']`, with this change arbitrary string will be valid. The actual problem is that currently `_type_check` is used for two different purposes:
* Check whether something is valid as a type
* Check whether something is valid as a type argument

The proper solution is to disentangle these two. Namely, at this particular position we only need to check if it is a valid type, so we can use a less restrictive check. I think the proper solution would be to add an optional argument to `_type_check` to skip the two corresponding checks.